src/igmpproxy.c

```diff
@@ -38,8 +38,10 @@
 */
 
 /* getopt() and clock_getime() */
-#ifndef _POSIX_C_SOURCE
-#define _POSIX_C_SOURCE 200112L
+#ifndef __FreeBSD__
+    #ifndef _POSIX_C_SOURCE
+    #define _POSIX_C_SOURCE 200112L
+    #endif
 #endif
```

Comment:
This change is suspicous. _POSIX_C_SOURCE 200112 is needed for getopt() and clock_getime() functions. I guess this #ifndef is some hack to workaround some problem. It should be fixed properly. Moreover it is not related to timing code rewrite, so should be in seprate commit.

Reply:
issue #35

Reply:
This change is some nonsense hack. Any explanation why it is needed? And of course should not be part of change which touches timing code.

Reply:
removed, but it is still needed to prevent these errors
/usr/include/net/route.h:82:2: error: unknown type name 'u_long'; did you mean 'long'?
        u_long  rmx_locks;      /* Kernel must leave these values alone */
        ^
/usr/include/net/route.h:83:2: error: unknown type name 'u_long'; did you mean 'long'?
        u_long  rmx_mtu;        /* MTU for this path */
        ^

Reply:
So it looks like you need to include some system specific header file prior to route.h...

Reply:
yes, I figured that out already, it has to do with order of including types.h netiniet/in.h etc.